napari/_qt/qt_plugin_list.py

```diff
@@ -0,0 +1,199 @@
+    """Item in a QtHookImplListWidget."""
+
+    def __init__(self, hookimpl, parent=None):
+        self.plugin_name = hookimpl.plugin_name
+        if self.plugin_name == 'builtins':
+            self.plugin_name = 'napari built-in'
+        super().__init__(self.plugin_name, parent)
+        self.hookimpl = hookimpl
+        self.setBackground(Qt.black)
+
+
+class QtHookImplListWidget(QListWidget):
+    """A QListWidget that allows sorting of plugin hooks."""
+
+    order_changed = Signal(list)  # emitted when the user changes the order.
+
+    def __init__(
+        self,
+        hook: Optional[_HookCaller] = None,
+        parent: Optional[QWidget] = None,
+    ):
+        """A ListWidget that shows all of the hook implementations for a spec.
+
+        Usually instantiated by QtPluginSorter.
+
+        Parameters
+        ----------
+        hook : pluggy.manager._HookCaller, optional
+            A pluggy HookCaller to show implementations for. by default None
+            (i.e. no hooks shown)
+        parent : QWidget, optional
+            Optional parent widget, by default None
+        """
+        super().__init__(parent)
+        self.setDefaultDropAction(Qt.MoveAction)
+        self.setDragEnabled(True)
+        self.setDragDropMode(self.InternalMove)
+        self.setAcceptDrops(True)
+        self.setSpacing(1)
+        self.setMinimumHeight(1)
+        self.setSizePolicy(
+            QSizePolicy.MinimumExpanding, QSizePolicy.MinimumExpanding
+        )
+        self.order_changed.connect(self.permute_hook)
+        self.set_hook(hook)
+
+    def set_hook(self, hook: Optional[_HookCaller]) -> None:
+        """Set the list widget to show hook implementations for ``hook``.
+
+        Parameters
+        ----------
+        hook : _HookCaller, optional
+            A pluggy HookCaller to show implementations for. by default None
+            (i.e. no hooks shown)
+        """
+        self.clear()
+        self.hook = hook
+        if not hook:
+            return
+        for hookimpl in reversed(hook.get_hookimpls()):
+            self.addItem(QtHookImplItem(hookimpl))
+
+    def dropEvent(self, event):
+        """Triggered when the user moves & drops one of the items in the list.
+
+        Parameters
+        ----------
+        event : QEvent
+            The event that triggered the dropEvent.
+        """
+        super().dropEvent(event)
+        order = [self.item(r).hookimpl for r in range(self.count())]
+        self.order_changed.emit(order)
+
+    def permute_hook(self, order: HookOrderType):
+        """Rearrage the call order of the hooks for the current hook impl.
+
+        Parameters
+        ----------
+        order : list
+            A list of str, hookimpls, or module_or_class, with the desired
+            CALL ORDER of the hook implementations.
+        """
+        if not self.hook:
+            return
+        permute_hook_implementations(self.hook, order)
+
+
+class QtPluginSorter(QDialog):
```

Comment:
This is great - as we add more hook specs / see how more plugins work, we might want to revisit, provide more options etc, but for now works well